transports/websocket-websys/src/web_context.rs

```diff
@@ -0,0 +1,62 @@
+use wasm_bindgen::{prelude::*, JsCast};
+
+/// Web context that abstract the window vs web worker API
+#[derive(Debug)]
+pub(crate) enum WebContext {
+    Window(web_sys::Window),
+    Worker(web_sys::WorkerGlobalScope),
+}
+
+impl WebContext {
+    pub(crate) fn new() -> Self {
+        #[wasm_bindgen]
+        extern "C" {
+            type Global;
+
+            #[wasm_bindgen(method, getter, js_name = Window)]
+            fn window(this: &Global) -> JsValue;
+
+            #[wasm_bindgen(method, getter, js_name = WorkerGlobalScope)]
+            fn worker(this: &Global) -> JsValue;
+        }
+
+        let global: Global = js_sys::global().unchecked_into();
+
+        if !global.window().is_undefined() {
+            Self::Window(global.unchecked_into())
+        } else if !global.worker().is_undefined() {
+            Self::Worker(global.unchecked_into())
+        } else {
+            panic!("Only supported in a browser or web worker");
+        }
+    }
+
+    /// The `setInterval()` method.
+    pub(crate) fn set_interval_with_callback_and_timeout_and_arguments(
+        &self,
+        handler: &::js_sys::Function,
+        timeout: i32,
+        arguments: &::js_sys::Array,
+    ) -> Result<i32, JsValue> {
+        match self {
+            WebContext::Window(w) => w.set_interval_with_callback_and_timeout_and_arguments(
+                handler,
+                timeout, // Chosen arbitrarily and likely worth tuning. Due to low impact of the /ws transport, no further effort was invested at the time.
```

Comment:
I don't think this comment makes any sense here because we are just passing on the variable.